lib/Sema/DebuggerTestingTransform.cpp

```diff
@@ -0,0 +1,245 @@
+//===--- DebuggerTestingTransform.cpp - Transform for debugger testing ----===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file This transform inserts instrumentation which the debugger can use to
+/// test its expression evaluation facilities.
+///
+//===----------------------------------------------------------------------===//
+
+#include "swift/AST/ASTContext.h"
+#include "swift/AST/ASTNode.h"
+#include "swift/AST/ASTWalker.h"
+#include "swift/AST/Decl.h"
+#include "swift/AST/DeclContext.h"
+#include "swift/AST/Expr.h"
+#include "swift/AST/Module.h"
+#include "swift/AST/ParameterList.h"
+#include "swift/AST/Stmt.h"
+#include "swift/Subsystems.h"
+
+#include "TypeChecker.h"
+
+using namespace swift;
+
+namespace {
+
+/// Find available closure discriminators.
+///
+/// The parser typically takes care of assigning unique discriminators to
+/// closures, but the parser is unavailable to this transform.
```

Comment:
can you add a little visual example that oultines what the transformation does, so it is easier to understand for the casual reader?

Reply:
I can't find a good way to visualize what this AST walker does, as it doesn't transform the AST. Do you have a rough outline / description of a visual example I could flesh out?